src/cose/nss.rs

```diff
@@ -157,26 +166,41 @@ fn hash(payload: &[u8]) -> Result<Vec<u8>, NSSError> {
 /// signed data.
 pub fn verify_signature(
     signature_algorithm: &SignatureAlgorithm,
-    spki: &[u8],
+    cert: &[u8],
     payload: &[u8],
     signature: &[u8],
 ) -> Result<(), NSSError> {
+    let slot = unsafe { PK11_GetInternalSlot() };
+    if slot.is_null() {
+        println!("Couldn't get the internal NSS slot.");
+        return Err(NSSError::LibraryFailure);
+    }
+
     let hash_buf = hash(payload).unwrap();
     let hash_item = SECItem::maybe_new(hash_buf.as_slice())?;
 
-    let spki_item = SECItem::maybe_new(spki)?;
-    // TODO: helper/macro for pattern of "call unsafe function, check null, defer unsafe release"?
-    let spki_handle = unsafe { SECKEY_DecodeDERSubjectPublicKeyInfo(&spki_item) };
-    if spki_handle.is_null() {
-        return Err(NSSError::DecodingSPKIFailed);
+    // Import DER cert into NSS.
+    let ee_cert_name = CString::new("ee_cert").unwrap();
+    let der_cert = SECItem::maybe_new(cert)?;
+    let db_handle = unsafe { CERT_GetDefaultCertDB() };
+    if db_handle.is_null() {
+        // TODO #28
+        return Err(NSSError::LibraryFailure);
+    }
+    let nss_cert = unsafe {
+        CERT_NewTempCertificate(db_handle, &der_cert, ee_cert_name.as_ptr(), false, false)
```

Comment:
I think the nickname may actually be optional here - that might simplify things a bit if we want to omit it.
Also, let's pass copyDER as true - otherwise we've got a potential uaf when der_cert goes out of scope, right?

Reply:
The `derCert` in the `nss_cert` isn't used I think. But we can copy it to be safe.
The nickname is optional indeed.